src/config.rs

```diff
@@ -66,18 +66,20 @@ impl Config {
         let contents = try!(Config::load_from_file(&path));
 
         let mut parser = Parser::new(&contents);
+        let parsed: Option<Table> = parser.parse();
 
-        let toml: Option<Table> = parser.parse();
-        let sphinx_toml = toml.and_then(|mut table| table.remove("package"))
+        // Verify parsed TOML is valid.
+        let mut toml: Table = try!(parsed.ok_or(FatalError::InvalidCargoFileFormat));
+
+        let config: Table = toml.remove("package")
```

Comment:
The truly dark and sneaky bit is below in the `unwrap_or(Table::new())` instead of fighting for `Option<Table>` in the `struct`. :)